app/controllers/v2/tailorings_controller.rb

```diff
@@ -43,9 +43,14 @@ def tailored?
     end
 
     def xccdf_tailoring_file
+      tailoring_rules = {}
+      (tailoring.security_guide.rules + tailoring.profile.rules).each do |rule|
+        tailoring_rules[rule.ref_id] = tailoring.rules.include?(rule)
+      end
+
```

Comment:
I think this computation should be extracted into the model, this runs at least 3 queries and does comparisons that can be done in SQL in a much more performant way.